### PR TITLE
[dashboards] Implement support for sunburst widget

### PR DESCRIPTION
**Pull request by DrkSephy**
This pull request implements support for the sunburst widget in Terraform.

### CODE REVIEW

File: docs/resources/dashboard.md [hunk 1]
```diff
@@ -7120,98 +7097,100 @@ Optional:
...
+- **right_yaxis** (Block List, Max: 1) A nested block describing the right Y-Axis Controls. See the `on_right_yaxis` property for which request will use this axis. The structure of this block is described below. (see [below for nested schema](#nestedblock--widget--group_definition--widget--timeseries_definition--right_yaxis))
+- **show_legend** (Boolean) Whether or not to show the legend on this widget.
+- **title** (String) The title of the widget.
+- **title_align** (String) The alignment of the widget's title. Valid values are `center`, `left`, `right`.
+- **title_size** (String) The size of the widget's title (defaults to 16).
+- **yaxis** (Block List, Max: 1) A nested block describing the Y-Axis Controls. The structure of this block is described below (see [below for nested schema](#nestedblock--widget--group_definition--widget--timeseries_definition--yaxis))
```
Comment (alai97) [inline docs/resources/dashboard.md]:
```suggestion
- **yaxis** (Block List, Max: 1) A nested block describing the Y-Axis Controls. The structure of this block is described below. (see [below for nested schema](#nestedblock--widget--group_definition--widget--timeseries_definition--yaxis))
```

File: docs/resources/dashboard.md [hunk 2]
```diff
@@ -2102,6 +2103,7 @@ Optional:
 - **scatterplot_definition** (Block List, Max: 1) The definition for a Scatterplot widget. (see [below for nested schema](#nestedblock--widget--group_definition--widget--scatterplot_definition))
 - **service_level_objective_definition** (Block List, Max: 1) The definition for a Service Level Objective widget. (see [below for nested schema](#nestedblock--widget--group_definition--widget--service_level_objective_definition))
 - **servicemap_definition** (Block List, Max: 1) The definition for a Service Map widget. (see [below for nested schema](#nestedblock--widget--group_definition--widget--servicemap_definition))
+- **sunburst_definition** (Block List, Max: 1) The definition for a Sunburst widget (see [below for nested schema](#nestedblock--widget--group_definition--widget--sunburst_definition))
```
Comment (alai97) [inline docs/resources/dashboard.md]:
```suggestion
- **sunburst_definition** (Block List, Max: 1) The definition for a Sunburst widget. (see [below for nested schema](#nestedblock--widget--group_definition--widget--sunburst_definition))
```

File: docs/resources/dashboard.md [hunk 3]
```diff
@@ -764,6 +764,7 @@ Optional:
 - **scatterplot_definition** (Block List, Max: 1) The definition for a Scatterplot widget. (see [below for nested schema](#nestedblock--widget--scatterplot_definition))
 - **service_level_objective_definition** (Block List, Max: 1) The definition for a Service Level Objective widget. (see [below for nested schema](#nestedblock--widget--service_level_objective_definition))
 - **servicemap_definition** (Block List, Max: 1) The definition for a Service Map widget. (see [below for nested schema](#nestedblock--widget--servicemap_definition))
+- **sunburst_definition** (Block List, Max: 1) The definition for a Sunburst widget (see [below for nested schema](#nestedblock--widget--sunburst_definition))
```
Comment (alai97) [inline docs/resources/dashboard.md]:
```suggestion
- **sunburst_definition** (Block List, Max: 1) The definition for a Sunburst widget. (see [below for nested schema](#nestedblock--widget--sunburst_definition))
```

**Review by alai97 — APPROVED** [review]
This is a hefty PR with big doc impact! These changes look great, suggested three minor period nits 👍

**Comment by DrkSephy** [comment]
@alai97 thanks for the review! Note that the docs you see here are autogenerated using `make docs` so the easiest way to review would be to look at all of the `description` keys inside of `datadog/resource_datadog_dashboard.go`. I've gone ahead and updated the punctuation in said descriptions and regenerated the docs.

File: datadog/resource_datadog_dashboard.go [hunk 4]
```diff
@@ -5009,6 +5025,376 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
...
+		},
+		"hide_total": {
+			Description: "Whether or not to show the total value in the widget.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+		},
+		"legend_inline": {
+			Description: "Used to configure the inline legend. Cannot be used in conjunction with legend_table.",
+			Type:        schema.TypeList,
+			Optional:    true,
```
Comment (skarimo) [inline datadog/resource_datadog_dashboard.go:5050]:
```suggestion
			Optional:    true,
			MaxItems:  1,
```

File: datadog/resource_datadog_dashboard.go [hunk 5]
```diff
@@ -5009,6 +5025,376 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
...
+						Type:        schema.TypeBool,
+						Optional:    true,
+					},
+				},
+			},
+		},
+		"legend_table": {
+			Description: "Used to configure the table legend. Cannot be used in conjunction with legend_inline.",
+			Type:        schema.TypeList,
+			Optional:    true,
```
Comment (skarimo) [inline datadog/resource_datadog_dashboard.go:5075]:
```suggestion
			Optional:    true,
			MaxItems:  1,
```

File: datadog/resource_datadog_dashboard.go [hunk 6]
```diff
@@ -5009,6 +5025,376 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
...
+		},
+		"legend_inline": {
+			Description: "Used to configure the inline legend. Cannot be used in conjunction with legend_table.",
+			Type:        schema.TypeList,
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"type": {
+						Description:      "The type of legend (table or none).",
```
Comment (skarimo) [inline datadog/resource_datadog_dashboard.go:5054]:
```suggestion
						Description:      "The type of legend.",
```

The auto generated docs should have the enums listed and I don't think `none` is valid in this case.

File: datadog/resource_datadog_dashboard.go [hunk 7]
```diff
@@ -5009,6 +5025,376 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
...
+			},
+		},
+		"legend_table": {
+			Description: "Used to configure the table legend. Cannot be used in conjunction with legend_inline.",
+			Type:        schema.TypeList,
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"type": {
+						Description:      "The type of legend (automatic or inline).",
```
Comment (skarimo) [inline datadog/resource_datadog_dashboard.go:5079]:
```suggestion
						Description:      "The type of legend.",
```
I think `none` is valid here.

File: datadog/resource_datadog_dashboard.go (hunk 4 above)
Comment (skarimo) [reply datadog/resource_datadog_dashboard.go:5050]: We can also add conflictsWith property here to make sure both are not set

Comment (DrkSephy) [reply datadog/resource_datadog_dashboard.go:5050]: My understanding is that `conflictsWith` will not work here due to the details in [this ticket](https://github.com/hashicorp/terraform-plugin-sdk/issues/71#issuecomment-666613275). Am I mistaken?

File: datadog/resource_datadog_dashboard.go (hunk 6 above)
Comment (DrkSephy) [reply datadog/resource_datadog_dashboard.go:5054]: `"none"` is a valid value for this field.

File: datadog/resource_datadog_dashboard.go (hunk 7 above)
Comment (DrkSephy) [reply datadog/resource_datadog_dashboard.go:5079]: I had these flipped, good catch!

**Review by alai97 — APPROVED** [review]

File: datadog/resource_datadog_dashboard.go (hunk 4 above)
Comment (skarimo) [reply datadog/resource_datadog_dashboard.go:5050]: 👍  fair point. We can keep it this way.

**Pull request merged**